lib/LoginStrategy.ts

```diff
@@ -239,4 +253,49 @@ export class LoginStrategy extends BaseStrategy {
       return acc
     }, [] as Claim[])
   }
-}
+
+  /**
+   * @description checks if the claimer is authenticated into the DID Document
+   *
+   * @param token
+   * @param payload
+   *
+   * @returns {string} issuer DID or null
+   */
+  async verifyDidClaim(token: string, payload: ITokenPayload): Promise<string> {
+
+    const didDocument = await this.didResolver.read(payload.iss)
+
+    if (this.isAuthorized(token, didDocument))
+      return payload.iss
+    return null 
+  }
+
+  private isAuthorized = (claimedToken: string,didDocument: IDIDDocument) : boolean => {
+
+    //retrieve claim infos from token
+    const claimInfos = jwt.decode(claimedToken) as {[key: string]: any;}
+    const didPubKey = didDocument.publicKey
+
+    if (didPubKey.length === 0)
+      return false
+    
+    //extract all authenticated users and keep those refering issuer's publicKey
+    const publicKeys = didDocument.authentication.map((auth, index) => {
+      const auth_pubkeyRef = auth["publicKey"] //public key reference in authentication field
+      const pubkey_id = didPubKey[index]["id"] //public key Id in publickKey field of DID document
+
+      /* If the auth field refers to the this public key entry, 
+        get the real public key in the publicKeyHex field */
+
+      if (extractId(auth_pubkeyRef) === extractId(pubkey_id)){
```

Comment:
`extractId` function isn't needed. The `id` is the entire string, not just the tag after the `#`

Reply:
For this I had a concern about the fact that the reference in the authentication field differs, after the '#' character, from the one in publickey field. So for now I extracted the parts before the #, which are similar in both fields.